examples/next/pages/ui/hooks/useDataStore/sort/index.page.tsx

```diff
@@ -0,0 +1,56 @@
+import { Amplify, DataStore, Hub } from 'aws-amplify';
+import { Authenticator } from '@aws-amplify/ui-react';
+import * as React from 'react';
+import '@aws-amplify/ui-react/styles.css';
+
+import { AuthSignOutButton } from '../AuthActions';
+import awsExports from '../aws-exports';
+import Collection from './Collection';
+import { initializeTestData } from '../utils';
+
+Amplify.configure({
+  ...awsExports,
+  aws_appsync_graphqlEndpoint: 'https://fake-appsync-endpoint/graphql',
+});
+
+export default function App() {
+  const [isInitialized, setInitialized] = React.useState(false);
+  const initializeStarted = React.useRef(false);
+
+  React.useEffect(() => {
+    (window as any).LOG_LEVEL = 'DEBUG';
+    Hub.listen('ui', (data) => {
+      console.log(data);
+    });
+    Hub.listen('datastore', (data) => {
+      console.log(data);
+    });
+    const initializeTestUserData = async () => {
+      if (initializeStarted.current) {
+        return;
+      }
+      await DataStore.clear();
+      await initializeTestData();
+      setInitialized(true);
+    };
+
+    initializeTestUserData();
+    initializeStarted.current = true;
```

Comment:
I don't have a good answer to why but the async init logic is called twice resulting in duplicate data without the initialized state.

Reply:
I believe this is a `React.StrictMode` thing:  https://reactjs.org/docs/strict-mode.html#ensuring-reusable-state

> This new check will automatically unmount and remount every component, whenever a component mounts for the first time, restoring the previous state on the second mount.

Can we do some cleanup in `useEffect` return statement to clean those data up so that data on second mount are clean?

Reply:
I did look into that but since DataStore.clear returns a promise, we have to await it, didn't work in return statement.